SDL/Constants.h

```diff
@@ -139,11 +139,11 @@ namespace SDL {
   const unsigned int size_superbins = 45000;
 
   // Temporary fix for endcap buffer allocation.
-  const unsigned int endcap_size = 9105;
+  const unsigned int endcap_size = 9104;
 
   // Temporary fix for module buffer allocation.
   const unsigned int modules_size = 26401;
-  const unsigned int pix_tot = 1796504;
+  const unsigned int pix_tot = 1794686;
```

Comment:
What about this change? Doesn't seem similar to the one explain in the comment above.

Also, are these still "temporary fixes"? Temporary in which sense?

Reply:
We commented at the same time, but to repeat what I said in my comment the pix_tot variable changes slightly because the CSV geometry is slightly different than the hit-based geometry, which leads to slightly different pixel maps with a different number of connections.

These are temporary fixes in the sense that they should be changed if possible (bug prone and annoying). It's really not the best solution, and goes back to these being global variables. Because they are global variables the sizes of them have to be defined at the very start of code execution (or compile time? I forget) according to Alpaka, so that means you have to manually have the sizes of the maps in as constants like above. CUDA didn't have this requirement, so in the old CUDA code the size was calculated dynamically during runtime and you didn't need these constants. The end result is that if you want to change the pixel maps right now you have to put your new maps in, run the code and let it crash with a warning I put in the code in a different file, and then update the pix_tot size with the size that the warning tells you to update it with. Not ideal.

Reply:
On the first point, I guess this could be another thing that changes the physics performance? I assume that timing remains unchanged?

On the second point, should we make this an issue?

Reply:
Yes but the change in physics performance from the maps alone is just a very tiny improvement in a few bins. I showed some plots in another comment under this PR where I show a comparison with the original buggy endcap map and everything else with the CSV geometry and those plots are almost identical to the comparison plots between CSV/old-maps. 

@VourMa I agree about making an issue for this, but maybe we wait for Slava's PR to be merged first? I think he moves around the globals in his PR, so maybe this can be fixed.

Reply:
Left a [note](https://github.com/SegmentLinking/TrackLooper/pull/377#issuecomment-2002643786) there.